src/Cardano/WalletLayer/Wallet.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE TypeOperators #-}
+
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides internal implementation of
+-- | wallet api
+
+module Cardano.WalletLayer.Wallet
+    (
+      CreateWallet
+    , WalletId
+    ) where
+
+import Prelude
+
+import Cardano.Wallet
+    ( Wallet )
+import Cardano.Wallet.AddressDerivation
+    ( Passphrase )
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap )
+import Cardano.Wallet.Mnemonic
+    ( Mnemonic )
+import Data.Text
+    ( Text )
+import Data.Time.Units
+    ( Microsecond )
+import GHC.Generics
+    ( Generic )
+
+-- | Errors
+data CreateWalletError
+
+data GetWalletError
+
+-- | Types
+data CreateWallet = CreateWallet NewWallet | ImportWallet
```

Comment:
This is the same thing in our new backend :)

Reply:
expanded `ImportWallet` to `ImportWallet (Key 'AccountK XPub)`